lib/docs-renderer/src/renderer.rs

```diff
@@ -0,0 +1,444 @@
+            // time, based on using `token` as the map key. If there's no map at the given key,
+            // we'll create one. If there's something other than a map, we'll panic.
+            let mut destination = map;
+            while let Some(segment) = segments.pop_front() {
+                match destination.get_mut(segment) {
+                    Some(Value::Object(ref mut next)) => {
+                        destination = next;
+                        continue;
+                    }
+                    Some(_) => panic!("Only leaf nodes should be allowed to be non-object values."),
+                    // If the next segment doesn't exist, there's nothing for us to delete, so return `false`.
+                    None => return false,
+                }
+            }
+
+            destination.remove(key).is_some()
+        })
+    }
+
+    /// Gets whether or not a value at the given path.
+    ///
+    /// The path follows the form of `/part1/part/.../partN`, where each slash-separated segment
+    /// represents a nested object within the overall object hierarchy. For example, a path of
+    /// `/root/nested/key2` would map to the value "weee!" if applied against the following JSON
+    /// object:
+    ///
+    ///   { "root": { "nested": { "key2": "weee!" } } }
+    ///
+    /// # Panics
+    ///
+    /// If the path does not start with a forward slash, this method will panic.
+    pub fn exists(&self, path: &str) -> bool {
+        if !path.starts_with('/') {
+            panic!("Paths must always start with a leading forward slash (`/`).");
+        }
+
+        // The root path always exists.
+        if path == "/" {
+            return true;
+        }
+
+        self.root.pointer(path).is_some()
+    }
+
+    /// Merges the data from `other` into `self`.
+    ///
+    /// Uses a "deep" merge strategy, which will recursively merge both objects together. This
+    /// strategy behaves as follows:
+    ///
+    /// - strings, booleans, integers, numbers, and nulls are "highest priority wins" (`self` has
+    ///   highest priority)
+    /// - arrays are merged together without any deduplication, with the items from `self` appearing
+    ///   first
+    /// - objects have their properties merged together, but if an overlapping property is
+    ///   encountered:
+    ///   - if it has the same type on both sides, the property is merged normally (using the
+    ///     standard merge behavior)
+    ///   - if it does not have the same type on both sides, the property value on the `self` side
+    ///     takes precedence
+    ///
+    /// The only exception to the merge behavior above is if an overlapping object property does not
+    /// have the same type on both sides, but the type on the `self` side is an array. When the type
+    /// is an array, the value on the `other` side is appended to that array, regardless of the
+    /// contents of the array.
+    pub fn merge(&mut self, _other: Self) {
+        todo!()
```

Comment:
Saving this until it's needed?

Reply:
Yeah.

There's likely some synergy, too, with the `Mergeable` trait I added when doing the work for the unevaluated properties visitor, so I need to think through that a little more.